specifyweb/frontend/js_src/lib/components/Statistics/StatsSpec.tsx

```diff
@@ -606,7 +618,7 @@ const statSpecToItems = (
 
 function generateBackEndSpec(statsSpec: StatsSpec): RA<{
   readonly responseKey: string;
-  readonly querySpec: QuerySpec | undefined;
+  readonly querySpec: ((dynamicResult: string) => QuerySpec) | undefined;
```

Comment:
careful, there may be an infinite re-render loop or at least a performance consideration from this - if before it was relying on the object being, but now the object would be different each time

maybe not an issue - you know your code best, so double check

Reply:
Not infinite re-render loops - the QuerySpec aren sent down to children components (and not to the parents). Also there are no useEffects that use this as a dependency. The performance problem is a real problem though. 

Fixed in  https://github.com/specify/specify7/pull/3942/commits/8d9c44df904202e253a97abbffe77d73fffa219b. The logic there is that queries which we generate for the user (like dynamic ones) can be cached - it already caches the other default ones (defined in stats spec) and the custom stats (coming from layout directly) (until user modifies them). When users do modify a dynamic stat (which they can), we don't care about getting a stale value from this ref (which we will), since it is automatically converted to custom stat with latest query specs.

Reply:
> beast with the stats page

It took me a while to re-understand how it works :)